src/server/graphql/mutations/__tests__/stripeSucceedPayment.test.js

```diff
@@ -0,0 +1,43 @@
+import DynamicSerializer from 'dynamic-serializer';
+import MockDB from 'server/__tests__/setup/MockDB';
+import fetchAndSerialize from 'server/__tests__/utils/fetchAndSerialize';
+import stripe from 'server/billing/stripe';
+import stripeWebhookHandler from 'server/billing/stripeWebhookHandler';
+import getRethink from 'server/database/rethinkDriver';
+import invoicePaymentSucceededEvent from 'server/graphql/mutations/__tests__/mockStripeEvents/invoicePaymentSucceededEvent';
+import shortid from 'shortid';
+import {PRO} from 'universal/utils/constants';
+import MockRes from 'server/__mocks__/MockRes';
+import MockReq from 'server/__mocks__/MockReq';
+
+console.error = jest.fn();
+
+describe('stripeSucceedPayment', () => {
+  test('handles invoice.payment_succeeded', async () => {
+    // SETUP
+    const invoiceId = `in_${shortid.generate()}`;
+    const res = new MockRes();
+    const r = getRethink();
+    const dynamicSerializer = new DynamicSerializer();
+    const mockDB = new MockDB();
+    const {organization} = await mockDB
+      .init({plan: PRO})
+      .newInvoice({id: invoiceId});
+    const org = organization[0];
+    stripe.__setMockData(org);
+    const req = new MockReq({body: invoicePaymentSucceededEvent(invoiceId, org.stripeId, org.stripeSubscriptionId)});
+    const subscription = stripe.__db.subscriptions[org.stripeSubscriptionId];
```

Comment:
A lot of code in this setup seems to be repeated throughout several of the test cases. If it's easily factorable w/ a reasonable name, it could make these tests a bit less verbose.

Reply:
when i look at it, the largest savings would only be 2 or 3 lines, which probably isn't worth it for an abstraction. if i'm mistaken feel free to clean it up, always a fan of cleaner code!

Reply:
Fair enough! If it's not easy to factor out logic with a nice name, then it's usually not worth it.